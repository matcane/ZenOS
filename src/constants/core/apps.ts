import calc from "@/config/apps/calc.json";
import camera from "@/config/apps/camera.json";
import chrome from "@/config/apps/chrome.json";
import clock from "@/config/apps/clock.json";
import messages from "@/config/apps/messages.json";
import phone from "@/config/apps/phone.json";
import settings from "@/config/apps/settings.json";

type TNavigationBar = {
  backgroundColor: string;
  dark: { backgroundColor: string };
};

export type TApp = {
  name: string;
  slug: string;
  version: string;
  icon: string;
  navigationBar?: TNavigationBar;
};

export const APPS = [settings, calc, clock];

export const PINNED_APPS = [phone, messages, chrome, camera];

export const APPS_ICON: Record<string, any> = {
  "@assets/icons/icons8-settings.webp": require("@assets/icons/icons8-settings.webp"),
  "@assets/icons/icons8-calculator.webp": require("@assets/icons/icons8-calculator.webp"),
  "@assets/icons/icons8-clock.webp": require("@assets/icons/icons8-clock.webp"),
  "@assets/icons/icons8-phone.webp": require("@assets/icons/icons8-phone.webp"),
  "@assets/icons/icons8-messages.webp": require("@assets/icons/icons8-messages.webp"),
  "@assets/icons/icons8-chrome.webp": require("@assets/icons/icons8-chrome.webp"),
  "@assets/icons/icons8-camera.webp": require("@assets/icons/icons8-camera.webp"),
};

export const APPS_ICON_SOURCE: Record<string, any> = {
  "@assets/icons/icons8-settings.webp": "https://icons8.com/icon/4UC7V4crVOgk/settings",
  "@assets/icons/icons8-calculator.webp": "https://icons8.com/icon/w8uA2lyOFNbd/calculator",
  "@assets/icons/icons8-clock.webp": "https://icons8.com/icon/kZYJCScZSxUH/clock",
  "@assets/icons/icons8-phone.webp": "https://icons8.com/icon/8U0YyPNowmTt/phone",
  "@assets/icons/icons8-messages.webp": "https://icons8.com/icon/U2maxZvgoDog/messages",
  "@assets/icons/icons8-chrome.webp": "https://icons8.com/icon/QBljOALaVp01/chrome",
  "@assets/icons/icons8-camera.webp": "https://icons8.com/icon/eKEI8q7Hjioq/camera",
};
