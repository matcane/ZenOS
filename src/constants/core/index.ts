export * from "./apps";
export * from "./settingsFields";
