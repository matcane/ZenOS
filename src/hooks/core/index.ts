export * from "./useTheme";
export * from "./useTime";
export * from "./useTimer";
