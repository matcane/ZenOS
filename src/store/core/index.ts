export * from "./themeStore";
export * from "./authStore";
export * from "./wallpaperStore";
