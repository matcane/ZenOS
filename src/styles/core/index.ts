import { StyleSheet } from "react-native";

import { baseStyle } from "../baseStyle";

export const coreStyles = StyleSheet.create({
  calcDisplay: {
    ...baseStyle.itemsEnd,
    ...baseStyle.justifyEnd,
    ...baseStyle.paddingBottomXL,
  },

  calcDisplayText: { ...baseStyle.paddingHorizontalMD },

  stopwatchStartStop: { right: "50%", marginRight: -30, bottom: 0, marginBottom: 30 },

  stopwatchReset: { right: "25%", marginRight: -20, bottom: 0, marginBottom: 35 },

  actionButton: {
    ...baseStyle.absolute,
    ...baseStyle.roundedFull,
    ...baseStyle.justifyCenter,
    ...baseStyle.itemsCenter,
    elevation: 5,
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },

  container: {
    ...baseStyle.paddingXL,
    ...baseStyle.itemsCenter,
  },

  navigationBarContainer: {
    ...baseStyle.justifyEvenly,
    ...baseStyle.itemsCenter,
    ...baseStyle.flexRow,
    height: 45,
    gap: 20,
  },

  backNavButton: { transform: [{ rotate: "30deg" }] },

  settingsFieldItem: {
    ...baseStyle.paddingLeftMD,
    ...baseStyle.flexRow,
    ...baseStyle.justifyBetween,
    ...baseStyle.itemsCenter,
    ...baseStyle.flexGrow,
    height: 60,
    gap: 10,
  },

  firstSettingsFieldItem: {
    borderTopEndRadius: 10,
    borderTopStartRadius: 10,
  },

  lastSettingsFieldItem: {
    borderBottomEndRadius: 10,
    borderBottomStartRadius: 10,
  },

  settingsFieldIcon: {
    ...baseStyle.sizeLG,
  },

  statusBarContainer: {
    ...baseStyle.absolute,
    ...baseStyle.flexRow,
    ...baseStyle.justifyBetween,
    ...baseStyle.itemsCenter,
    ...baseStyle.paddingHorizontalMD,
    height: 32,
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    zIndex: 999,
  },

  statusBarIcons: {
    ...baseStyle.flexRow,
    gap: 2,
  },

  wallpaper: {
    ...baseStyle.flexGrow,
    ...baseStyle.justifyCenter,
    ...baseStyle.paddingTopXL,
  },
});
