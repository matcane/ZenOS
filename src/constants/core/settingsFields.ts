import { TApp } from "./apps";

export type TSettingField = Omit<TApp, "slug" | "version" | "navigationBar"> & {
  toggle?: boolean;
  path?: string;
};

export const ACCOUNT_FIELDS = [
  { name: "Phone", icon: "@assets/icons/icons8-messages.webp", path: "settings/modals/Phone" },
];

export const CUSTOMIZATION_FIELDS = [
  { name: "Theme", icon: "@assets/icons/icons8-paint-palette.webp", path: "settings/modals/Theme" },
];

export const APPS_FIELDS = [
  { name: "App", icon: "@assets/icons/icons8-apps.webp", path: "settings/modals/Apps" },
];

export const OS_FIELDS = [{ name: "Shutdown", icon: "" }];

export const SETTINGS_ICON: Record<string, any> = {
  "@assets/icons/icons8-apps.webp": require("@assets/icons/icons8-apps.webp"),
  "@assets/icons/icons8-paint-palette.webp": require("@assets/icons/icons8-paint-palette.webp"),
  "@assets/icons/icons8-messages.webp": require("@assets/icons/icons8-messages.webp"),
};
