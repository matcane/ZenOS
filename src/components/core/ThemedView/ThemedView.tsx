import { View, ViewProps } from "react-native";

import { useTheme } from "@/hooks/core";

export default function ThemedView({ style, ...rest }: ViewProps) {
  const theme = useTheme();
  const backgroundColor = theme.background;
  return <View style={[{ backgroundColor }, style]} {...rest} />;
}
