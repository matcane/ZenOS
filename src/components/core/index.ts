import ActionButton from "./ActionButton/ActionButton";
import AppWidget from "./AppWidget/AppWidget";
import AuthForm from "./AuthForm/AuthForm";
import CalcButton from "./CalcButton/CalcButton";
import NavigationBar from "./NavigationBar/NavigationBar";
import SafeArea from "./SafeArea/SafeArea";
import SettingsField from "./SettingsField/SettingsField";
import StatusBar from "./StatusBar/StatusBar";
import ThemedText from "./ThemedText/ThemedText";
import ThemedView from "./ThemedView/ThemedView";
import Wallpaper from "./Wallpaper/Wallpaper";

export {
  NavigationBar,
  StatusBar,
  ThemedText,
  ThemedView,
  Wallpaper,
  ActionButton,
  AppWidget,
  CalcButton,
  SettingsField,
  SafeArea,
  AuthForm,
};
