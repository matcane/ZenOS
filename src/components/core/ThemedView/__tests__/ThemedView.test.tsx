import { render } from "@testing-library/react-native";
import { Text } from "react-native";

import { useTheme } from "@/hooks/core";

import ThemedView from "../ThemedView";

jest.mock("@/hooks/core", () => ({
  useTheme: jest.fn(),
}));

describe("ThemedView component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    (useTheme as jest.Mock).mockReturnValue({ background: "#fff" });
    const { toJSON } = render(<ThemedView />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("applies the correct background color from the theme", () => {
    const mockTheme = { background: "#f0f0f0" };
    (useTheme as jest.Mock).mockReturnValue(mockTheme);

    const { getByTestId } = render(<ThemedView testID="themed-view" />);

    const themedView = getByTestId("themed-view");
    expect(themedView.props.style).toContainEqual({ backgroundColor: mockTheme.background });
  });

  it("applies additional styles passed via props", () => {
    (useTheme as jest.Mock).mockReturnValue({ background: "#fff" });

    const { getByTestId } = render(<ThemedView style={{ padding: 10 }} testID="themed-view" />);

    const themedView = getByTestId("themed-view");
    expect(themedView.props.style).toContainEqual({ padding: 10 });
  });

  it("renders children correctly", () => {
    (useTheme as jest.Mock).mockReturnValue({ background: "#fff" });

    const { getByText } = render(
      <ThemedView>
        <Text>Child content</Text>
      </ThemedView>,
    );

    expect(getByText("Child content")).toBeTruthy();
  });
});
