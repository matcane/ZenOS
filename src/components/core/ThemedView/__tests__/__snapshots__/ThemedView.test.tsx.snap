// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedView component renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
/>
`;
